integration/commands_diagnostic_test.go

```diff
@@ -406,3 +409,133 @@ func TestCommandsDiagnosticWhatsMyURI(t *testing.T) {
 		assert.NotEqual(t, ports[0], ports[1])
 	}
 }
+
+// TestCommandWhatsMyURIConnection tests that a client uses the same connection for
+// all commands run by the client, and different clients use different connection.
+// The port number is used to validate if the same connection is used or not,
+// since unique port number is assigned to each client connection in the test setup.
+func TestCommandWhatsMyURIConnection(t *testing.T) {
+	t.Parallel()
+
+	// set 1 to ensure only one pool exists duration of the test,
+	// which forces a client to use a single connection pool
+	q1 := url.Values{}
+	q1.Set("maxPoolSize", "1")
+	q1.Set("minPoolSize", "1")
+
+	s := setup.SetupWithOpts(t, &setup.SetupOpts{
+		ExtraOptions: q1,
+	})
+
+	collection1 := s.Collection
+	databaseName := s.Collection.Database().Name()
+	collectionName := s.Collection.Name()
+
+	t.Run("SameClient", func(t *testing.T) {
+		setup.SkipExceptMongoDB(t, "https://github.com/FerretDB/FerretDB/issues/2906")
+
+		t.Parallel()
+
+		num := runtime.GOMAXPROCS(-1) * 10
```

Comment:
Please use `teststress.Stress`

Reply:
Made minor change to export goroutine numbers.